### ISSUE TITLE
[BUG] Azure.Deployment.SecureValue false positive fail with resource function output 

### ISSUE THREAD

**Issue by holyyyns** [Azure/PSRule.Rules.Azure#2919]
### Existing rule

Azure.Deployment.SecureValue

### Description of the issue

When using any resource function other than listKeys() output as property value Azure.Deployment.SecureValue is returning false fail.

### Error messages

[FAIL] Azure.Deployment.SecureValue (AZR-000316)
    | Template: modules/search/searchservices-adminkey/test/main.test.bicep:0:0

    | RECOMMEND:
    | Consider using secure parameters for sensitive resource properties.

    | REASON:
    | - The property 'properties.value' must use a secure parameter but does not.
    | - The property 'properties.value' must use a secure parameter but does not.

    | HELP:
    | - https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.Deployment.SecureValue/

##[error]AZR-000316: /home/vsts/work/1/s/modules/search/searchservices-adminkey/test/main.test.bicep failed Azure.Deployment.SecureValue. Use secure parameters for setting properties of resources that contain sensitive information.

### Reproduction

```
// Resources
resource resSearchService 'Microsoft.Search/searchServices@2022-09-01' existing = {
  name: varResourceName
  scope: resourceGroup(parSearchServiceResourceGroupName)
}

resource resKeyVault 'Microsoft.KeyVault/vaults@2023-07-01' existing = {
  name: parKeyVaultName
}

resource resKeyVaultSecretSearchPrimaryAdminKey 'Microsoft.KeyVault/vaults/secrets@2023-07-01' = {
  parent: resKeyVault
  name: '${resSearchService.name}-primaryAdminKey'
  properties: {
    value: resSearchService.listAdminKeys().primaryKey
  }
}

resource resKeyVaultSecretSearchSecondaryAdminKey 'Microsoft.KeyVault/vaults/secrets@2023-07-01' = {
  parent: resKeyVault
  name: '${resSearchService.name}-secondaryAdminKey'
  properties: {
    value: resSearchService.listAdminKeys().secondaryKey
  }
}
```

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.37.0

### Additional context

listQueryKeys() does the same

**Comment by BernieWhite** [comment]
Thanks for reporting the issue @holyyyns
